clients/fides-js/src/integrations/gtm.ts

```diff
@@ -78,6 +67,23 @@ const pushFidesVariableToGTM = (
     });
   }
 
+  // Then override with actual consent values
+  if (consent) {
+    if (privacyNotices && flagType === GtmFlagType.CONSENT_MECHANISM) {
+      Object.entries(consent).forEach(([key, value]) => {
+        consentValues[key] = transformConsentToFidesUserPreference(
+          value,
+          privacyNotices.find((notice) => notice.notice_key === key)
+            ?.consent_mechanism,
+        );
+      });
```

Comment:
I'm wondering if we should take the time to extract a function to build the consent object that we pass an array of settings to. That function could compose the consent object from those arguments.